runner/internal/config/config.go

```diff
@@ -83,13 +84,34 @@ type Output struct {
 	Silent bool
 }
 
+type set map[string]struct{}
+
+func (set set) add(values ...string) {
+	for _, v := range values {
+		set[v] = struct{}{}
+	}
+}
+
 // Global represents the global configuration of the runner.
 // It must be validated using Global.Validate before usage.
 type Global struct {
 	Request Request
 	Runner  Runner
 	Output  Output
 	Tests   []tests.Case
+
+	fieldsSet set
+}
+
+// WithField returns a new Global with the input fields marked as set.
+func (cfg Global) WithFields(fields ...string) Global {
+	fieldsSet := cfg.fieldsSet
+	if fieldsSet == nil {
+		fieldsSet = set{}
+	}
+	fieldsSet.add(fields...)
+	cfg.fieldsSet = fieldsSet
```

Comment:
Question: is there a reason why we do not mutate directly `cfg.fieldsSet`?

```suggestion
	if cfg.fieldsSet == nil {
		cfg.fieldsSet = set{}
	}
	cfg.fieldsSet.add(fields...)
```